File: 백준/Bronze/14530. The Lost Cow/The Lost Cow.py
x, y = map(int, input().split())
dap = 0
dir = 1
move = 1
cur = x
while 1:
    next = x + dir * move
    if (x <= y <= next) or (next <= y <= x):
        dap += abs(y - cur)
        break
    else:
        dap += abs(next - cur)
        cur = next
        move *= 2
        dir *= -1
print(dap)

File: 백준/Bronze/14530. The Lost Cow/The Lost Cow.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();
        int y = sc.nextInt();
        int dap = 0;
        int dir = 1;
        int move = 1;
        int cur = x;
        while (true) {
            int next = x + dir * move;
            if ((x <= y && y <= next) || (next <= y && y <= x)) {
                dap += Math.abs(y - cur);
                break;
            } else {
                dap += Math.abs(next - cur);
                cur = next;
                move *= 2;
                dir *= -1;
            }
        }
        System.out.println(dap);
        sc.close();
    }
}